Initial 0.13.1 proposal

andreyluiz opened the pull request:
Hello.

This PR describes a proposal to migrate the tutorial to version v0.13.1.

Here are the topics covered in this PR:
- Migration of the markdown files to v0.13.1 (Fixes #12)
- Initialization of the repo as a binary crate with examples (Fixes #13)
- Not implemented yet, but update the readme file with historical versions (Fixes #1)
  - This means the current main branch would need to be pushed as v0.12.1, as you did with v0.10.0

If you like the proposal, I will continue working on this migration.

---

This is a TODO list for the migration.
- [x] adding_widgets.md
- [x] batch_tasks.md
- [x] batch_subscriptions.md
- [x] button.md
- [x] ~changing_displaying_content.md~ Renamed to application_lifecycle.md
- [x] changing_styles.md
- [x] changing_the_window_dynamically.md
- [x] changing_themes.md
- [x] checkbox.md
- [x] closing_the_window_on_demand.md
- [x] column.md
- [x] combobox.md
- [x] container.md
- [x] ~controlling_widgets_by_commands.md~ Renamed to controlling_widgets_behavior.md
- [x] custom_background.md
- [x] ~custom_styles.md~ Removed because `changing_styles` explains broadly
- [x] dispatching_tasks_at_startup.md **ADDED**
- [x] drawing_shapes.md
- [x] drawing_widgets.md
- [x] drawing_with_caches.md
- [x] executing_custom_tasks.md
- [x] ~explanation_of_sandbox_trait.md~ Renamed to explanation_of_app_structure.md
- [x] first_app.md
- [x] ~from_sandbox_to_application.md~ Application trait has been deprecated so this was removed
- [x] image.md
- [x] ~initializing_a_different_window.md~ Renamed to customizing_the_window_on_launch.md
- [x] loading_images_asynchronously.md
- [x] memoryless_pages.md
- [x] more_than_one_page.md
- [x] mouse_pointer_over_widgets.md
- [x] navigation_history.md
- [x] on_pressed_released_of_some_widgets.md
- [x] passing_parameters_across_pages.md
- [x] picklist.md
- [x] producing_messages_by_keyboard_events.md
- [x] producing_messages_by_mouse_events.md
- [x] producing_messages_by_timers.md
- [x] producing_widget_messages.md
- [x] progressbar.md
- [x] radio.md
- [x] row.md
- [x] rule.md
- [x] scrollable.md
- [x] setting_up.md
- [x] slider.md
- [x] space.md
- [x] svg.md
- [x] taking_any_children.md
- [x] text_input.md
- [x] text.md
- [x] texts_in_widgets.md
- [x] toggler.md
- [x] tooltip.md
- [x] updating_widgets_from_events.md
- [x] updating_widgets_from_outside.md
- [x] widgets_with_children.md
- [x] width_and_height.md

fogarecious commented:
Everything looks great to me! Just let me know whenever you're ready to merge the pull request.

andreyluiz commented:
Made some more migrations.
- Renamed "Explanation of Sandbox trait" to "Explanation of the App Structure"
- Added a Widgets page as a high level section in the tutorial
- Added a file to explain the application lifecycle

andreyluiz commented:
More migrations done:
- Finished migrating the widgets
- Added a Layouts page as a high level section in the tutorial
- Migrated the layouts

BlakeGardner commented:
Just wanted to say thanks for updating this, @andreyluiz, As an Iced newbie, I was having trouble getting the tutorial code to work, and this update fixed my issues.

andreyluiz commented:
Glad to help, @BlakeGardner. I will try to finish the whole migration this week. ;)

andreyluiz commented:
More migrations:
- Migrated the styles section
- Removed the custom styling from the styles section (not applicable anymore)
- Migrated the multipage apps section

andreyluiz commented:
Alright, this one took a little longer because I had to restructure a the tutorials for the tasks. Before we had commands, which were renamed to tasks. I tried to make the tasks section in a way that doesn't get too complex for people seeing it for the first time.

andreyluiz commented:
Migrate the windows section. I took a while because I had to figure out how to get an equivalence of `window::Id::MAIN`. By looking at the code of the Iced library, I saw that internally they use `window::get_latest()`.

- Migrate windows section
- Renamed initializing_a_different_window to customizing_the_window_on_launch

andreyluiz commented:
No big renames on these two:
- Migrated events section
- Migrated canvas section

andreyluiz commented:
And that's it!

- Migrated custom widgets section
- Migrated final section
- Added an end page

andreyluiz commented:
@fogarecious I also created the PR https://github.com/iced-rs/awesome-iced/pull/27 on awesome-iced to update the version of the tutorial. It will be merged after this one.

fogarecious commented:
Thank you so much! It looks fantastic to me.
